common/src/main/java/org/corfudb/common/metrics/micrometer/MeterRegistryProvider.java

```diff
@@ -0,0 +1,91 @@
+package org.corfudb.common.metrics.micrometer;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.logging.LoggingMeterRegistry;
+import io.micrometer.core.instrument.logging.LoggingRegistryConfig;
+import org.corfudb.common.metrics.micrometer.loggingsink.InfluxLineProtocolLoggingSink;
+import org.corfudb.common.metrics.micrometer.loggingsink.LoggingSink;
+import org.slf4j.Logger;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+/**
+ * A configuration class for a meter (metrics) registry.
+ */
+public class MeterRegistryProvider {
+    private static Optional<MeterRegistry> meterRegistry = Optional.empty();
+
+    private MeterRegistryProvider() {
+
+    }
+
+    /**
+     * Get the previously configured meter registry.
+     * If the registry has not been previously configured, create a default logging MeterRegistry
+     * and return.
+     * @return An optional configured meter registry.
+     */
+    public static Optional<MeterRegistry> getInstance() {
+        if (!meterRegistry.isPresent()) {
```

Comment:
here is a possible race. Imagine you have two threads
t1->getInstance()
t2->getInstance()

t1-> meterRegistry is not present
at the  same time
t12-> meterRegistry is not present ALSO

t1->creates LoggingMeterRegistry
t2->creates LoggingMeterRegistry

Please take a look https://www.journaldev.com/1377/java-singleton-design-pattern-best-practices-examples

I would prefer "Eager initialization" approach, if possible. 
If you would like to use your method you need to add an additional check, see "4. Thread Safe Singleton"